lib/opentelemetry/wrapper.ts

```diff
@@ -1,15 +1,57 @@
 import { getCurrentMeasurementScope, isRoute06User } from "@/app/(auth)/lib";
 import { waitUntil } from "@vercel/functions";
 import type { LanguageModelUsage } from "ai";
+import type { LanguageModelV1 } from "ai";
 import type { Strategy } from "unstructured-client/sdk/models/shared";
 import { captureError } from "./log";
 import type { LogSchema, OtelLoggerWrapper } from "./types";
 import {
 	ExternalServiceName,
 	type RequestCountSchema,
+	type TokenBasedServiceName,
 	type TokenConsumedSchema,
+	UnimplementedServiceName,
 } from "./types";
 
+type ModelInfo = {
+	externalServiceName: TokenBasedServiceName;
+	modelId: string;
+};
+
+function getModelInfo(
+	logger: OtelLoggerWrapper,
+	model: LanguageModelV1,
+): ModelInfo {
+	const modelClassName = model.constructor.name;
+
+	switch (modelClassName) {
+		case "OpenAIChatLanguageModel":
+			return {
+				externalServiceName: ExternalServiceName.OpenAI,
+				modelId: model.modelId,
+			};
+		case "AnthropicMessagesLanguageModel":
+			return {
+				externalServiceName: ExternalServiceName.Anthropic,
+				modelId: model.modelId,
+			};
+		case "GoogleGenerativeAILanguageModel":
+			return {
+				externalServiceName: ExternalServiceName.Google,
+				modelId: model.modelId,
+			};
+		default:
+			logger.error(
+				new Error(`unknown model class name '${modelClassName}' obtained`),
+				"consider adding to 'ExternalServiceName'",
+			);
+			return {
+				externalServiceName: UnimplementedServiceName.TODO,
```

Comment:
`UnimplementedServiceName.TODO` is defined to fulfill type safety, not for actual runtime use case.